torch/_inductor/utils.py

```diff
@@ -1538,6 +1538,15 @@ def num_fw_fixed_arguments(dynamo_gm_num_inputs: int, aot_fw_gm_num_inputs: int)
     num_rng_seed_offset_inputs = (
         2 if torch._functorch.config.functionalize_rng_ops else 0
     )
+    # AOT won't lift any parameters if we're inlining NN Modules
+    # however desugaring subclasses will still add arguments
+    # resulted in extra fixed inputs https://github.com/pytorch/pytorch/issues/130502
+    if (
+        torch._dynamo.config.inline_inbuilt_nn_modules
+        and not torch._dynamo.utils.is_parameter_freezing()
+    ):
+        return 0
+
     return aot_fw_gm_num_inputs - dynamo_gm_num_inputs - num_rng_seed_offset_inputs
```

Comment:
Not directly related to this PR but @angelayi effect tokens will cause AOTDispatcher to add extra arguments in the AOT graph that were not in the dynamo graph. We probably need to account for them here? (otherwise inductor will think that there are extra graphargs that have static addresses)

Reply:
If the backend is inductor, AOTDispatcher will not add extra arguments to the AOT graph. Instead it will insert a `          _make_dep_token` and `_sink_tokens` call in the graph (more specific graph pastes in https://github.com/pytorch/pytorch/pull/122347).

Reply:
ahhh right thank you!